### DISCUSSION TITLE
Allow "freetext" for ui.select's autocomplete

### DISCUSSION THREAD

**Discussion by groucho86** [zauberzeug/nicegui#1045] [Ideas / Feature Requests]
with_input + new option "freetext" would allow for users to select from options but also add their own values.

**Comment by falkoschindler** [comment]
Thanks, @groucho86, that might be useful. Although the result would be similar to [`ui.input` with autocompletion](https://nicegui.io/documentation/input#autocompletion).

**Reply by groucho86** [reply]
Similar yes, but ui.input with autocompletion implies the user knows what they're about to type, whereas ui.select (with freetext option) allows the user to see "suggestions" as well as input whatever they want.

**Reply by falkoschindler** [reply]
Ok, I see. 👍

**Comment by falkoschindler** [comment]
I guess we could use the `new-value-mode` prop. When a new value is sent to the server, the `ui.select` would need to detect this case in `_msg_to_value`:

```py
            if msg['args'] is None:
                return None
            elif isinstance(msg['args'], str):
                return msg['args']
            else:
                return self._values[msg['args']['value']]
```

Anyone up for sorting out the details and creating a pull request? 🙂

**Reply by falkoschindler** [reply]
See PR https://github.com/zauberzeug/nicegui/pull/1819

**Reply by falkoschindler** [reply]
PR #1819 has been released in version 1.4.2.
